Upstream-Status: Inappropriate [other]
Upstream is not making further releases of this software.

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

# Fix a problem with chmod attributes when using no_squash_all
# Patch origin: Wind River

---
 setattr.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/setattr.c
+++ b/setattr.c
@@ -115,7 +115,7 @@ nfsstat setattr(char *path, sattr *attr,
 		}
 	}
 
-	if (flags & SATTR_CHMOD) {
+	if (flags & SATTR_CHMOD && attr->mode != -1) {
 		unsigned int	mode = attr->mode;
 
 		/* If setuid is not allowed, silently squash them */
